resources/virt.go

```diff
@@ -120,12 +120,29 @@ func (obj *VirtRes) Validate() error {
 	return nil
 }
 
-func (obj *VirtRes) connect() (conn libvirt.VirConnection, err error) {
+func (obj *VirtRes) connect() (conn *libvirt.Connect, err error) {
 	if obj.Auth != nil {
-		conn, err = libvirt.NewVirConnectionWithAuth(obj.URI, obj.Auth.Username, obj.Auth.Password)
+		callback := func(creds []*libvirt.ConnectCredential) {
+			for _, cred := range creds {
```

Comment:
Since I'm unfamiliar with this new callback mechanism, is the idea to just modify the values passed in?? A small comment here or in the patch would be appreciated.